definitions/npm/react-redux_v5.x.x/flow_v0.89.x-/test_connect.js

```diff
@@ -0,0 +1,495 @@
+  };
+
+  const mapStateToProps = (state: {}) => ({
+    // $ExpectError wrong type for stringProp
+    stringProp: false,
+  });
+
+  const mapDispatchToProps = () => ({
+    // $ExpectError wrong type for numProp
+    numProp: false,
+  });
+
+  connect<Props, {||}, _,_,_,_>(mapStateToProps, mapDispatchToProps)(Component);
+}
+
+function doesNotRequireDefinedComponentToTypeCheck4case() {
+  type Props = {
+    stringProp: string,
+  };
+
+  const Component = ({ stringProp }: Props) => {
+    return <span>{stringProp}</span>;
+  };
+
+  const mapStateToProps = (state: {}) => ({
+    // $ExpectError wrong type for stringProp
+    stringProp: false,
+  });
+
+  connect<Props, {||}, _,_,_,_>(mapStateToProps, {})(Component);
+}
+
+function doesNotRequireDefinedComponentToTypeCheck5case() {
+  type Props = {
+    stringProp: string
+  };
+
+  const Component = ({ stringProp }: Props) => {
+    return <span>{stringProp}</span>;
+  };
+
+  const mapStateToProps = () => ({});
+  const mapDispatchToProps = () => ({});
+
+  const mergeProps = () => ({
+    // $ExpectError wrong type for stringProp
+    stringProp: true
+  });
+
+  connect(mapStateToProps, mapDispatchToProps, mergeProps)(Component);
+}
+
+function testExactProps() {
+  type Props = {|
+    forMapStateToProps: string,
+    passthrough: number,
+    fromStateToProps: string
+  |};
+
+  class Com extends React.Component<Props> {
+    render() {
+      return <div>{this.props.passthrough} {this.props.fromStateToProps}</div>;
+    }
+  }
+
+  type State = {a: number};
+  type InputProps = {|
+    forMapStateToProps: string,
+    passthrough: number,
+  |};
+
+  const mapStateToProps = (state: State, props: InputProps) => {
+    return {
+      fromStateToProps: 'str' + state.a
+    }
+  };
+
+  const Connected = connect(mapStateToProps)(Com);
+  <Connected passthrough={123} forMapStateToProps={'data'} />;
+  //$ExpectError extra prop what exact props does not allow
+  <Connected passthrough={123} forMapStateToProps={321} extraProp={123}/>;
+  //$ExpectError wrong type for forMapStateToProps
+  <Connected passthrough={123} forMapStateToProps={321}/>;
+  //$ExpectError passthrough missing
+  <Connected forMapStateToProps={'data'} />;
+  //$ExpectError forMapStateToProps missing
+  <Connected passthrough={123}/>;
+  //$ExpectError takes in only React components
+  connect(mapStateToProps)('');
+}
+
+function testWithStatelessFunctionalComponent() {
+  type Props = {passthrough: number, fromStateToProps: string};
+  const Com = (props: Props) => <div>{props.passthrough} {props.fromStateToProps}</div>
+
+  type State = {a: number};
+  type InputProps = {
+    forMapStateToProps: string
+  };
+  const mapStateToProps = (state: State, props: InputProps) => {
+    return {
+      fromStateToProps: 'str' + state.a
+    }
+  };
+
+  const Connected = connect(mapStateToProps)(Com);
+  <Connected passthrough={123} forMapStateToProps={'data'}/>;
+  //$ExpectError wrong type for passthrough
```

Comment:
complains about missing `fromStateToProps` instead of wrong type for `passthrough`